### ISSUE TITLE
Add benchmarking suite

### ISSUE THREAD

**Issue by mdboom** [matplotlib/matplotlib#2188]
It would be nice to have a standard suite of benchmarks -- particularly those things that are known performance bottlenecks -- so we can track their improvement or regression over time and provide data about the relative performance of different backends etc.

**Comment by petehuang** [comment]
Hi,

This issue has been inactive for a while. Are there any updates? Any initial ideas that we have to help people start thinking about this? Thanks!

**Comment by tacaswell** [comment]
To increase visibility tagged as 'new-contributor-friendly' as this is a very self-contained project, but medium difficulty as it will involve sorting out how to set up ASV and how to find reliable hardware to run it on (our typical CI tricks won't work here as we need fixed resources underneath).  Maybe pricing AWS for this?

**Comment by dopplershift** [comment]
Rackspace may also be an option; I think that's what powers https://speed.python.org/

**Comment by jklymak** [comment]
Having the suite alone would be a useful first step.  Automating it could come later.

**Comment by QuLogic** [comment]
I am working on this; will try to publish something this weekend.

**Comment by jklymak** [comment]
I'll bump the milestone though, as I don't think there is a higher rush.  Just getting into master should be enough.

**Comment by QuLogic** [comment]
We have [some benchmarks here](https://github.com/matplotlib/mpl-bench) with results that end up [here](https://matplotlib.org/mpl-bench/). Those results are produced on my machine and I try to run them while I'm not using it, but there's not real guarantee of independence there.

We are working to get it running on the same dedicated hardware used for Pandas benchmarking.

**Comment by anntzer** [comment]
Perhaps worth adding a benchmark on just the time to import matplotlib, or matplotlib.pyplot (but forcing the backend to agg, as that's going to be backend dependent).

**Comment by QuLogic** [comment]
There's a PR for that in asv: airspeed-velocity/asv#487 but it seems to have stalled.

**Comment by QuLogic** [comment]
Waiting on https://github.com/TomAugspurger/asv-runner/pull/2, though I think I need to update to asv 0.3 as well.
